frontend/src/store/slices/searchSlice.ts

```diff
@@ -0,0 +1,67 @@
+import { createAsyncThunk, createSlice } from "@reduxjs/toolkit";
+import type { RootState } from "..";
+
+export interface SearchResult {
+	type: "song" | "artist" | "playlist";
+	id: number;
+	name: string;
+	thumb_url: string | null;
+	artist_name: string | null;
+}
+
+interface SearchState {
+	results: SearchResult[];
+	isLoading: boolean;
+	error: string | null;
+}
+
+const initialState: SearchState = {
+	results: [],
+	isLoading: false,
+	error: null,
+};
+
+export const searchContent = createAsyncThunk(
+	"search/searchContent",
+	async (query: string) => {
+		if (!query || query.length < 2) return { results: [] };
+
+		const response = await fetch(`/api/search?q=${encodeURIComponent(query)}`);
```

Comment:
can we move these fetches and api definitions into api.ts so that the api directory remains consistent?

Reply:
Yes good call I will do this and re commit